stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/feature/messages/list/internal/poll/PollResultsDialogFragment.kt

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.ui.feature.messages.list.internal.poll
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatDialogFragment
+import androidx.appcompat.widget.Toolbar
+import androidx.core.content.ContextCompat
+import androidx.core.view.isVisible
+import androidx.recyclerview.widget.RecyclerView
+import io.getstream.chat.android.models.Poll
+import io.getstream.chat.android.models.Vote
+import io.getstream.chat.android.ui.ChatUI
+import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.ui.databinding.StreamUiFragmentPollResultsBinding
+import io.getstream.chat.android.ui.databinding.StreamUiItemResultBinding
+import io.getstream.chat.android.ui.databinding.StreamUiItemResultUserBinding
+import io.getstream.chat.android.ui.utils.extensions.streamThemeInflater
+
+/**
+ * Represent the bottom sheet dialog that allows users to pick attachments.
+ */
+public class PollResultsDialogFragment : AppCompatDialogFragment() {
+
+    private var _binding: StreamUiFragmentPollResultsBinding? = null
+    private val binding get() = _binding!!
+    private lateinit var poll: Poll
```

Comment:
What are your thoughts on moving this property inside the primary constructor? Alternatively, we could make the primary constructor private, ensuring that instances can only be created via the `newInstance` method.

```kotlin
public class PollResultsDialogFragment(
    private val poll: Poll
) : AppCompatDialogFragment() {
...
```

If not people can create `PollResultsDialogFragment` without passing the `poll` instance and get crashes.

```kotlin
val fragment = PollResultsDialogFragment()
```

Reply:
I second this, we should keep the default constructor.